tests/unit/source_code/test_jobs.py

```diff
@@ -512,3 +513,39 @@ def test_xxx(graph):
     assert workflow_task_container.spark_conf == {"spark.databricks.cluster.profile": "singleNode"}
 
     ws.assert_not_called()
+
+
+@pytest.mark.parametrize(
+    "name, query, dfsa_paths, is_read, is_write",
+    [
+        ("simple", "SELECT * from dual", [], False, False),
+        (
+            "location",
+            "CREATE TABLE hive_metastore.indices_historical_data.sp_500 LOCATION 's3a://db-gtm-industry-solutions/data/fsi/capm/sp_500/'",
+            ["s3a://db-gtm-industry-solutions/data/fsi/capm/sp_500/"],
+            False,
+            True,
+        ),
+    ],
+)
+def test_workflow_linter_collects_dfsas_from_queries(
+    name,
+    query,
+    dfsa_paths,
+    is_read,
+    is_write,
+    mock_path_lookup,
+    simple_dependency_resolver,
+    empty_index,
+    mock_dfsa_crawler,
+):
+    ws = create_autospec(WorkspaceClient)
+    query = Query.from_dict({"parent_path": "workspace", "display_name": name, "query_text": query})
+    response = ListQueryObjectsResponseQuery.from_dict(query.as_dict())
```

Comment:
@nfx the `Query.parent_path` field is not carried over to `ListQueryObjectsResponseQuery`... Should I file an issue with engineering ? How do I do that ?

Reply:
why do we need parent path there?

Reply:
Because 2 queries with the same name can be stored under 2 different folders (tested). As I understand it, parent_path contains the folder